google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java

```diff
@@ -1290,6 +1304,31 @@ public void onCompleted() {
     return result;
   }
 
+  public void getPartitions(long partitionCount, ApiStreamObserver<QueryPartition> observer) {
+    PartitionQueryRequest.Builder request = PartitionQueryRequest.newBuilder();
+    request.setPartitionCount(partitionCount);
+    request.setStructuredQuery(buildQuery());
+    request.setParent(options.getParentPath().toString());
+    FirestoreClient.PartitionQueryPagedResponse pagedResponse =
+        firestore.getClient().partitionQuery(request.build());
+
+    @Nullable Object[] lastCursor = null;
```

Comment:
I found the name of this variable confusing while reading the code, as I interpreted it to mean the last cursor for all the partitions. Can we change it to `previousCursor` instead to clarify the idea that it's a back reference?

Reply:
Sure thing, i will update this (soon)